extensions/wikia/BannerNotifications/js/BannerNotifications.js

```diff
@@ -0,0 +1,294 @@
+			.setContent($.msg('bannernotifications-general-ajax-failure'))
+			.show();
+	};
+
+	/**
+	 * Allows to attach a handler to a notification close event
+	 * @param {Function} callback
+	 * @returns {BannerNotification}
+	 */
+	BannerNotification.prototype.onClose = function (callback) {
+		if (typeof callback === 'function') {
+			this.onCloseHandler = callback;
+		}
+		return this;
+	};
+
+	/**
+	 * Called once to instantiate this feature
+	 */
+	function init() {
+		if (window.skin === 'monobook') {
+			$pageContainer = $('#content');
+			headerHeight = 0;
+		} else {
+			$pageContainer = $('.WikiaPageContentWrapper');
+			headerHeight = $('#globalNavigation').height();
+		}
+		createBackendNotification();
+		onScroll.bind(handleScrolling);
+	}
+
+	/**
+	 * Creates an instance of a notification sets on the backend side
+	 * (if such one exists)
+	 */
+	function createBackendNotification() {
+		var $backendNotification = $('.banner-notification');
+		if ($backendNotification.length) {
+			backendNotification = new BannerNotification();
+			setUpClose(backendNotification);
+		}
+	}
+
+	/**
+	 * Pins the notification to the top of the screen when scrolled down the page.
+	 */
+	function handleScrolling() {
+		var containerTop,
+			notificationWrapper = $pageContainer.children('.banner-notifications-wrapper');
+
+		if (!$pageContainer.length || !notificationWrapper.length) {
+			return;
+		}
+
+		// get the position of the wrapper element relative to the top of the viewport
+		containerTop = $pageContainer[0].getBoundingClientRect().top;
+
+		if (containerTop < headerHeight) {
+			notificationWrapper.addClass('float');
+		} else {
+			notificationWrapper.removeClass('float');
+		}
+	}
+
+	/**
+	 * Constructs jQuery element with the notification
+	 * @param {String} content
+	 * @param {String} [type]
+	 * @returns {jQuery}
+	 */
+	function createMarkup(content, type) {
+		return $(
+			mustache.render(templates.BannerNotifications, {
+				imageSource: closeImageSource,
+				content: content
+			})
+		).addClass(type).hide();
+	}
+
+	/**
+	 * Adds given markup to DOM
+	 * @param {jQuery} $element
+	 * @param {jQuery} $parentElement
+	 */
+	function addToDOM($element, $parentElement) {
+		// allow notification wrapper element to be passed by extension
+		var $parent = $parentElement || isModalShown() ? modal : $pageContainer,
+			$bannerNotificationsWrapper = $parent.find('.banner-notifications-wrapper');
+		if (!$bannerNotificationsWrapper.length) {
+			$bannerNotificationsWrapper = $('<div></div>').addClass('banner-notifications-wrapper');
+			$parent.prepend($bannerNotificationsWrapper);
+		}
+		$bannerNotificationsWrapper.prepend($element);
+
+		$element.fadeIn('slow');
+	}
+
+	/**
+	 * Bind close event to close button
+	 */
+	function setUpClose(notification) {
+		$(notification.$element).on('click', '.close', function (event) {
+			notification.hide();
+			if (notification.onCloseHandler) {
+				notification.onCloseHandler(event);
```

Comment:
what do you think about defining ```onClonseHandler``` as ```Function.prototype``` by default and just call it without checking if it's defined.